### PR TITLE
Number in-use traits in Piece Definer

### PR DESCRIPTION
**Pull request by riverwanderer**
Cleans up PR #12835.

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java [hunk 1]
```diff
@@ -1281,15 +1292,36 @@ public Component getListCellRendererComponent(
...
       if (value instanceof EditablePiece) {
-        setText(((EditablePiece) value).getDescription());
+        final EditablePiece editableValue = (EditablePiece) value;
+        final String type = editableValue.getType().split(";")[0];
+
+        // Prototype? Skip first (dummy) item to avoid outputting a blank line
+        if (lineOffset == 1 || index > 0) {
+          // For Pieces, bump index to 1
+          // Special formatting for Comments
+          if (type.startsWith("cmt")) {
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java:1306]: Better would be `if (type.startsWith(Commend.ID))`

**Review by BrentEaston — CHANGES_REQUESTED** [review]
Just one minor change. Otherwise looks great.

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java (hunk 1 above)
Comment (riverwanderer) [reply vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java]: Done, accounting for the trailing separator in Comment.ID.

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java [hunk 2]
```diff
@@ -1281,15 +1292,36 @@ public Component getListCellRendererComponent(
       // It is incredibly inefficient for GamePieces and is not needed
       // since we overwrite the label text anyway.
       super.getListCellRendererComponent(list, "", index, selected, hasFocus);
+
+      final int lineOffset = definer.isPrototype() ? 1 : 0;
+
       if (value instanceof EditablePiece) {
-        setText(((EditablePiece) value).getDescription());
+        final EditablePiece editableValue = (EditablePiece) value;
+        final String type = editableValue.getType().split(";")[0] + ";";
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java:1300]: Why is this needed? You can just check for if (editableValue.getType().startsWith(Comment.ID)). No need to break it up since the ; is already included in the ID.

**Review by BrentEaston — CHANGES_REQUESTED** [review]

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java (hunk 2 above)
Comment (riverwanderer) [reply vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java]: Thanks - fixed in next build.

**Review by BrentEaston — APPROVED** [review]

File: vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java [hunk 3]
```diff
@@ -1281,15 +1292,35 @@ public Component getListCellRendererComponent(
...
         final String s = value.getClass().getName();
-        setText(s.substring(s.lastIndexOf('.') + 1));
+        setText(getLineNumber(index + lineOffset) + s.substring(s.lastIndexOf('.') + 1) + "</html>");
       }
       return this;
     }
+
+    private static String getLineNumber(int lineNumber) {
+      return "<html><span style=color:#A0A0A0>" + lineNumber + ".</span> ";  // prep line number for output
+    }
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java:1323]: I would not produce the `<html>` tags unbalanced like this. The function name says it produces the line number, so it should do only that. It would be better to have matching open and close tags produced in the same place.

Comment (uckelman) [reply vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java:1323]: I suggest something like this:
```
private String cellText(int lineno, String s) {
  return "<html>" + getLineNumber(lineno) + s + </html>";
}
```
and then
```
setText(cellText(index + lineOffset, "whatever"));
```
